vllm/lora/peft_helper.py

```diff
@@ -38,6 +38,8 @@ def _validate_features(self):
 
     def __post_init__(self):
         self._validate_features()
+        if self.use_rslora:
+            self.vllm_scaling_factor = self.lora_alpha / math.sqrt(self.r)
         if self.context_length:
             if self.vllm_max_position_embeddings is None:
                 self.vllm_max_position_embeddings = self.context_length
```

Comment:
@jeejeelee I think this is the cleanest way to support the RSLoRA scaling logic! My main outstanding worry is I am not sure how the scaling applied for RSLoRA and the scaling applied when `self.context_length` is provided interact.

Reply:
> My main outstanding worry is I am not sure how the scaling applied for rsLoRA and the scaling applied when self.context_length is provided should interact.

IMHO, These two `scaling` are distinct and operate independently of each other.

We shouldn't use `vllm_scaling_factor `- instead, we can add a variable called `scaling`

Reply:
Fine by me, but its gets a little confusing as `LoRAModel` already defines `scaling_factor` and uses it for long context support:

```python
class LoRAModel(AdapterModel):
    """A LoRA fine-tuned model."""

    def __init__(
        self,
        lora_model_id: int,
        rank: int,
        loras: Dict[str, LoRALayerWeights],
        scaling_factor: Optional[float] = None,
    ) -> None:
        """
        Args:
            lora_model_id: The integer id for the lora model.
            rank: lora rank.
            loras: module name -> weights for lora-replaced layers.
            scaling_factor: Scaling factor to support long context lora model.
                None if the lora is not tuned for long context support.
```

Is there an argument to sync `LoRAModel` and `PEFTHelper` so each has a `scaling_factor` and `vllm_scaling_factor` argument?

Reply:
`long_context_scaling_factor ` looks reasonable. Let me verify this tomorrow and confirm

Reply:
I have implemented the related logic, please check if it's reasonable

Reply:
This looks good to me @jeejeelee! Thanks for adding tests

Reply:
Could you please sync with the main branch?

Reply:
Done. It is just the DCO check that is failing but I am unsure how to fix, it suggests a rebase but I don't want to do that as there's multiple commit authors on this branch

Reply:
There is a known issue in the current lora test. Let's check if other lora tests can pass first. If they do, we can consider force merging